### PR TITLE
[Validator] support `\Stringable` instances in all constraints

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | 
| License       | MIT

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Validator/Constraints/PasswordStrengthValidator.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ public function validate(#[\SensitiveParameter] mixed $value, Constraint $constr
             return;
         }
 
-        if (!\is_string($value)) {
+        if (!\is_string($value) && !$value instanceof \Stringable) {
             throw new UnexpectedValueException($value, 'string');
         }
         $passwordStrengthEstimator = $this->passwordStrengthEstimator ?? self::estimateStrength(...);
```
Comment (GromNaN) [inline src/Symfony/Component/Validator/Constraints/PasswordStrengthValidator.php:42]: The value may be casted to `(string)` when calling `$passwordStrengthEstimator($value)`. We don't know how this is handled by the `$passwordStrengthEstimator` callable.

**Review by GromNaN — APPROVED** [review]

**Comment by GromNaN** [comment]
It would be better with tests.

**Comment by xabbuh** [comment]
@GromNaN tests added

File: src/Symfony/Component/Validator/Constraints/PasswordStrengthValidator.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/Validator/Constraints/PasswordStrengthValidator.php:42]: indeed, done

**Review by OskarStark — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**